packages/core/contracts/oracle/implementation/SkinnyOptimisticOracle.sol

```diff
@@ -0,0 +1,716 @@
+
+    /**
+     * @notice Generates stamped ancillary data in the format that it would be used in the case of a price dispute.
+     * @param _ancillaryData ancillary data of the price being requested.
+     * @param _requester sender of the initial price request.
+     * @return the stamped ancillary bytes.
+     */
+    function stampAncillaryData(bytes memory _ancillaryData, address _requester)
+        public
+        pure
+        override
+        returns (bytes memory)
+    {
+        return _stampAncillaryData(_ancillaryData, _requester);
+    }
+
+    // Returns hash of unique request identifiers. This contract maps request ID hashes to hashes of the request's
+    // parameters.
+    function _getId(
+        address _requester,
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData
+    ) private pure returns (bytes32) {
+        return keccak256(abi.encode(_requester, _identifier, _timestamp, _ancillaryData));
+    }
+
+    // Returns hash of request parameters. These are mapped to the unique request ID to track a request's lifecycle.
+    function _getRequestHash(Request memory _request) private pure returns (bytes32) {
+        return keccak256(abi.encode(_request));
+    }
+
+    // Resolves a price request that has expired or been disputed and a price is available from the DVM. This will
+    // revert if the unique request ID does not match the hashed request parameters. This also marks the request
+    // as settled, therefore this method can only be triggered once per eligible request.
+    function _settle(
+        address _requester,
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData,
+        Request memory _request
+    ) private returns (uint256 payout, int256 resolvedPrice) {
+        bytes32 requestId = _getId(_requester, _identifier, _timestamp, _ancillaryData);
+        _validateRequestHash(requestId, _request);
+
+        // Associate settled request params with ID.
+        Request memory settledRequest =
+            Request({
+                proposer: _request.proposer,
+                disputer: _request.disputer,
+                currency: _request.currency,
+                settled: true, // Modified
+                proposedPrice: _request.proposedPrice,
+                resolvedPrice: _request.resolvedPrice,
+                expirationTime: _request.expirationTime,
+                reward: _request.reward,
+                finalFee: _request.finalFee,
+                bond: _request.bond,
+                customLiveness: _request.customLiveness
+            });
+
+        OptimisticOracleInterface.State state =
+            _getState(_requester, _identifier, _timestamp, _ancillaryData, _request);
+        if (state == OptimisticOracleInterface.State.Expired) {
+            // In the expiry case, just pay back the proposer's bond and final fee along with the reward.
+            resolvedPrice = settledRequest.proposedPrice;
```

Comment:
question: is there a gas difference between using `settledRequest.proposedPrice` and `request.proposedPrice`?

Reply:
Looks like it saves < 100 gas i'll make the change though

Reply:
Yup I made all the changes!